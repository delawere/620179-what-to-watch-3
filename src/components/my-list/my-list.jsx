// Libs
import React, {PureComponent} from "react";
import {array, func} from 'prop-types';
import {connect} from "react-redux";
// Utils
import {Operation as FavoritesOpearion} from "../../reducer/favorites/favorites.js";
import {getFavorites} from "../../reducer/favorites/selectors.js";
// Components
import Logo from "../logo/logo.jsx";
import Footer from "../footer/footer.jsx";
import Avatar from '../avatar/avatar.jsx';
import MovieCard from "../movie-card/movie-card.jsx";

class MyList extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {loadFavorites} = this.props;
    loadFavorites();
  }

  render() {
    const {favorites = []} = this.props;

    return (
      <div className="user-page">
        <header className="page-header user-page__head">
          <Logo />

          <h1 className="page-title user-page__title">My list</h1>

          <div className="user-block">
            <Avatar />
          </div>
        </header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <div className="catalog__movies-list">
            {favorites.map((film) => {
              const {id} = film;
              return <MovieCard key={id} {...film}/>;
            })}
          </div>
        </section>

        <Footer />
      </div>
    );
  }
}

MyList.propTypes = {
  favorites: array,
  loadFavorites: func
};

const mapStateToProps = (state) => ({
  favorites: getFavorites(state)
});

const mapDispatchToProps = (dispatch) => ({
  loadFavorites() {
    dispatch(FavoritesOpearion.loadFavorites());
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
