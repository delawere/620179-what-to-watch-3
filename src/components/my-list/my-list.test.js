import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {MyList} from "./my-list.jsx";

const props = {
  favorites: [
    {
      id: 1,
      name: `test`
    }
  ],
  loadFavorites: () => {}
};

it(`MyList renders correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <MyList {...props} />
      </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
