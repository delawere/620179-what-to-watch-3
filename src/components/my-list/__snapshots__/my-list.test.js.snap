// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="test"
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "height": "100%",
                "width": "100%",
              }
            }
          >
            test
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
